### ISSUE TITLE
Keyboard navigation additional tests

### ISSUE THREAD

**Issue by zdrawku** [IgniteUI/igniteui-angular#2705]
### Description  
Implement all missing test scenarios based on the Wiki keyboard navigation [test specification](https://github.com/IgniteUI/igniteui-angular/wiki/igxGrid-Specification#keyboard-navigation-scenarios)

Push the tests to branch [gridKeyboardNav](https://github.com/IgniteUI/igniteui-angular/tree/ddincheva/gridKeyboardNav), related to [PR](https://github.com/IgniteUI/igniteui-angular/pull/2677).
